### PR TITLE
[#14] Add property tests

### LINKED ISSUE #14: Add property tests
# Clarification and motivation

Despite its simplicity, and the unit tests should be sufficient, it would be great to add some property tests to this contract.

# Acceptance criteria

Add property tests that verify the contract behavior based on randomized inputs.

### PR DESCRIPTION
**Pull request by lierdakil**
This only actually adds what's basically one complicated property test, because I'm struggling to come up with other interesting properties we could test for besides "some invariants hold during contract operation", hence it's a single scenario.

~~Draft since based off (and depends on) #11~~

## Description

Problem: We only have unit tests.

Solution: Add a randomized property test for the main workflow path
of the contract.

## Related issue(s)

Resolves #14 

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

**Review by rinn7e — APPROVED** [review]
Looks good to me 👍

File: package.yaml [hunk 1]
```diff
@@ -136,3 +136,7 @@ tests:
     - fmt
     - morley-ledgers
     - data-default
+    - hedgehog
+    - tasty-hedgehog
+    - o-clock
+    - containers
```
Comment (pasqu4le) [inline package.yaml:142]: nitpick, but can you keep this list sorted please?

**Review by pasqu4le — APPROVED** [review]
LGTM :ok_hand:

File: package.yaml (hunk 1 above)
Comment (lierdakil) [reply package.yaml:142]: Yes.

**Pull request merged**